source/common/http/header_map_impl.h

```diff
@@ -324,6 +324,19 @@ class ResponseTrailerMapImpl : public HeaderMapImpl, public ResponseTrailerMap {
   friend class HeaderMapImpl;
 };
 
+class HeaderListView {
+public:
+  using HeaderKeyValuePairWrapper =
+      std::vector<std::pair<std::reference_wrapper<const HeaderString>,
+                            std::reference_wrapper<const HeaderString>>>;
+  HeaderListView(const HeaderMapImpl& header_map);
```

Comment:
this should not take an impl. It should take a public header map. Also, please move this to a different file/utility that does not depend on header_map_impl